scripts/utils/auroLibraryUtils.mjs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+// ---------------------------------------------------------------------
+
+import * as fs from 'fs';
+import * as path from 'path';
+import chalk from 'chalk';
+
+export default class AuroLibraryUtils {
+
+  /**
+   * Copys and pastes all files in a source directory into a destination directory.
+   * @param {String} srcDir - File path of directory to copy from.
+   * @param {String} destDir - File path of directory to paste files into.
+   * @param {Boolean} removeFiles - If true, removes all files in destination directory before pasting files.
+   */
+  copyDirectory(srcDir, destDir, removeFiles) {
+    // Removes all files from directory
+    if (removeFiles && fs.existsSync(destDir)) {
+      const destFiles = fs.readdirSync(destDir);
+
+      let filesRemoved = 0;
+      let filesFailed = 0;
+  
+      destFiles.forEach(file => {
+        try {
+          const filePath = path.join(destDir, file);
+          fs.unlinkSync(filePath);
+          this.auroLogger(`Removed file: ${file}`);
+
+          filesRemoved += 1;
+        } catch(err) {
+          this.auroLogger(`Failed to remove ${file}: ${err}`, 'error');
+          filesFailed += 1;
+        }
+      });
+
+      if (filesRemoved > 0) {
+        this.auroLogger(`Removed ${filesRemoved} files`, 'success');
+      }
+
+      if (filesFailed > 0) {
+        this.auroLogger(`Failed to remove ${filesRemoved} files`, 'error');
+      }
+    }
+
+    // Creates destination directory if it does not exist
+    if (!fs.existsSync(destDir)) {
+      fs.mkdirSync(destDir);
+    }
+  
+    // All files from source directory
+    const files = fs.readdirSync(srcDir);
+  
+    // Copys over all files from source directory to destination directory
+    files.forEach(file => {
+      const sourceFilePath = path.join(srcDir, file);
+      const destFilePath = path.join(destDir, file);
+  
+      const stat = fs.statSync(sourceFilePath);
+  
+      if (stat.isDirectory()) {
+        this.copyDirectory(srcDir, destDir, removeFiles);
+      } else {
+        try {
+          fs.copyFileSync(sourceFilePath, destFilePath);
+          this.auroLogger(`Copied file: ${file}`, 'success');
+        } catch (err) {
+          this.auroLogger(`Error copying file ${file}: ${err}`, 'error');
```

Comment:
No I was not able to test the logger. The way that the code is written is that if there is no existing destination directory, it creates that directory as well.

Reply:
What about giving it a bad source directory/file path?

Reply:
When giving a bad source directory path, an error is thrown in the console, but not from the `auroLogger`. Here is an example output of what comes back from an incorrect file path.
<img width="1091" alt="Screen Shot 2023-08-23 at 11 27 35 AM" src="https://github.com/AlaskaAirlines/auro-library/assets/55807669/0fd887fe-8190-44e3-ad65-43fbcc03ab7a">

Reply:
To answer your question, I do not believe the `catch` branch is ever reached because the error shown in the screenshot is thrown before the catch function is ever reached.